# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserBridgeHost.cpp
# Commit: 1663719e2aa5
# Full Hash: 1663719e2aa5b6f77e32398c6d6fcb4cb3c999ed
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 16 - Cycle collect RemoteBrowser and support weak references in BrowserHost. r=nika
#   
#   BrowserParent is cycle collected and supported weak references, so this commit adds support
#   for these things to BrowserHost.
#   
# ==============================================================================

diff -r 53663cad8890 -r 1663719e2aa5 dom/ipc/BrowserBridgeHost.cpp
--- a/dom/ipc/BrowserBridgeHost.cpp	Wed May 08 16:12:26 2019 -0500
+++ b/dom/ipc/BrowserBridgeHost.cpp	Tue May 14 16:28:10 2019 -0500
@@ -9,6 +9,15 @@
 namespace mozilla {
 namespace dom {
 
+NS_INTERFACE_MAP_BEGIN_CYCLE_COLLECTION(BrowserBridgeHost)
+  NS_INTERFACE_MAP_ENTRY(nsISupports)
+NS_INTERFACE_MAP_END
+
+NS_IMPL_CYCLE_COLLECTION(BrowserBridgeHost)
+
+NS_IMPL_CYCLE_COLLECTING_ADDREF(BrowserBridgeHost)
+NS_IMPL_CYCLE_COLLECTING_RELEASE(BrowserBridgeHost)
+
 BrowserBridgeHost::BrowserBridgeHost(BrowserBridgeChild* aChild)
     : mBridge(aChild) {}
 